pipeline/authn/authenticator_oauth2_introspection.go

```diff
@@ -96,6 +98,35 @@ type AuthenticatorOAuth2IntrospectionResult struct {
 	TokenUse  string                 `json:"token_use"`
 }
 
+func (a *Audience) UnmarshalJSON(b []byte) error {
```

Comment:
I mean the behavior is quite obvious, but please still add a test for this function to prevent regression issues when someone comes and changes that later on.

Reply:
I'll add a test for both the string and the array case.

Reply:
@zepatrik I added some tests. What do you think?